EIPS/eip-7701.md

```diff
@@ -194,6 +299,177 @@ abi.encode(uint256 actualGasCost, bool success, bytes context)
+
+For all existing transaction types, during the single top-level execution frame,
+the `ORIGIN` (`0x32`, `tx.origin`) and `CALLER` (`0x33`, `msg.sender`)
+are both equal the address that is determined by the transaction's ECDSA signature (`yParity`, `r`, `s`).
+
+When processing an EIP-7701 transaction, however, multiple execution frames will be created.
+The full list of possible frames and their corresponding code sections is as follows:
+
+1. **Validation Phase**
+    * `sender` deployment frame (once per account) - `role_sender_deployment`
+    * `sender` validation frame (required) - `role_sender_validation`
+    * `paymaster` validation frame (optional) - `role_paymaster_validation`
+2. Execution Phase
+    * `sender` execution frame (required) - `role_sender_execution`
+    * `paymaster` post-transaction frame (optional) - `role_paymaster_post_tx`
+
+All execution frames in the **Validation Phase** must be completed successfully without reverting,
+and both `sender` and `paymaster` validation frames must include a call to a corresponding `AA_ENTRY_POINT`
+approval callback functions in order for the transaction to be considered valid for a given position in a block.
+
+In all top-level frames, the global variables have the following meaning:
+
+| Opcode Name | Solidity Equivalent | Value                                                                         |
+|-------------|---------------------|-------------------------------------------------------------------------------|
+| `CALLER`    | `msg.sender`        | The `AA_ENTRY_POINT` address. `AA_SENDER_CREATOR` for the "deployment frame". |
+| `ORIGIN`    | `tx.origin`         | The transaction `sender` address                                              |
+| `CALLDATA*` | `msg.data`          | The transaction data is set to inputs of the corresponding frame              |
+
+### Transaction execution context
+
+Note that some behaviours in the EVM depend on the transaction context. These behaviours are:
+
+1. Costs of the `SSTORE (0x55)` opcode per [EIP-2200](./eip-2200)
+2. Costs of accessing cold addresses and slots per [EIP-2929](./eip-2929)
+3. Values available within the transient storage per [EIP-1163](./eip-1163)
+4. Maximum amount of gas refund assigned after the execution per [EIP-3529](./eip-3529)
+
+These features are not affected by the separation of the transaction into multiple frames.
+Meaning, for example, that a value set with `TSTORE (0x5D)` in one frame will remain available in the next one.
+
+### Flow diagrams
+
+#### Simple AA Transaction flow
+
+![Simple AA transaction flow diagram](../assets/eip-7701/simple_flow.svg)
+
+#### Complete AA transaction flow
+
+![Simple AA transaction flow diagram](../assets/eip-7701/complete_flow.svg)
+
+### Pseudocode
+
+#### AA transaction state transition function
+
+```python
+
+def state_transition_function(tx):
+    buy_gas(tx)
+    deployer_result = deployer_frame(tx)
```

Comment:
Please add checks for deployer frame and paymaster frame since they're optional, and it's not obvious

Reply:
I did this by returning `{success: True}` inside the `deployer_frame` function, to keep the `state_transition_function` as clean as possible.